errdefs/helpers_test.go

```diff
@@ -130,3 +166,29 @@ func TestIsDataLoss(t *testing.T) {
 		t.Fatalf("causual should be errTest, got: %v", cause)
 	}
 }
+
+func TestUnavailable(t *testing.T) {
+	if IsUnavailable(errTest) {
+		t.Fatalf("did not expect unavaillable error, got %T", errTest)
+	}
+	e := Unavailable(errTest)
+	if !IsUnavailable(e) {
+		t.Fatalf("expected unavaillable error, got %T", e)
+	}
+	if cause := e.(causal).Cause(); cause != errTest {
+		t.Fatalf("causual should be errTest, got: %v", cause)
+	}
+}
+
+func TestSystem(t *testing.T) {
+	if IsSystem(errTest) {
```

Comment:
This test is failing; I left it in, because I'm not sure why it fails

Reply:
Still not sure why this is matching the generic error

Reply:
🤦‍♂️ Look at the `ErrSystem` inteface

Reply:
doh! Looked at it and didn’t spot that; https://github.com/moby/moby/blob/3a633a712c8bbb863fe7e57ec132dd87a9c4eff7/errdefs/defs.go#L38

`ErrSystem()` should be `System()` :joy:

Reply:
updated 👍